### DISCUSSION TITLE
New version

### DISCUSSION THREAD

**Discussion by rk-yen** [openuado/niet#74] [General]
Hi @4383 !

First of all, thank you for providing the extremely helpful package !

Question:
Do you have any plans on releasing a new version to pypi, especially after the update(s) to fix the vulnerabilities in pyyaml (Ref: #68 )?

**Comment by 4383** [comment]
Hello,

Unfortunately our automatic deployment pipeline have been broken by recent travis updates, so, I've to release it manually.
I'll try to release a new version early next week.

**Comment by 4383** [comment]
Done!

Niet 2.5.0 is now out https://pypi.org/project/niet/2.5.0/

**Reply by rk-yen** [reply]
> Done!
> 
> 
> 
> Niet 2.5.0 is now out https://pypi.org/project/niet/2.5.0/

Thank you for the quick response!!!

**Comment by rk-yen** [comment]
Hi @4383 ... looks like something went wrong with the metadata.
```
$ pip show niet

Name: niet
Version: 0.0.0
Summary: Shell config file parser (json, yaml)
Home-page: https://github.com/openuado/niet/
Author: openuado
Author-email: herveberaud.pro@gmail.com
License: MIT
Location: /Users/raviy/.venvs/foo/lib/python3.9/site-packages
Requires: 
Required-by: 
```

```
Collecting niet==2.5.0
  Downloading https://<private_url>/niet/2.5.0/niet-2.5.0.tar.gz (28 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  WARNING: Requested niet==2.5.0 from <snip> <snip> requires-python:>=3.6): Requested niet==2.5.0 from https://<private_url>/niet/2.5.0/niet-2.5.0.tar.gz#sha256=d75f939d731e61ee58e7b576856ed8b0b1b7607095974a94ca13b8e4a9a7a9b4 (from -r requirements-test.txt (line 2)) has inconsistent version: expected '2.5.0', but metadata has '0.0.0'
ERROR: Could not find a version that satisfies the requirement niet==2.5.0 (from versions: 0.1.0, 0.2.0, 0.2.1, 0.2.2, 1.0.0, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.7.0, 1.8.0, 1.8.1, 1.8.2, 2.0.0, 2.1.0, 2.2.0, 2.3.0, 2.4.0, 2.5.0)
ERROR: No matching distribution found for niet==2.5.0
Error: Process completed with exit code 1.
```

**Comment by rk-yen** [comment]
Hi @4383 ! 
Can you please take another pass at this at your earliest convenience? Thanks in advance.

**Comment by 4383** [comment]
Hi @rk-yen,

Sorry for my late reply.
Yes I'll try to release a new version with the right version ASAP.

Thanks for the heads up.

**Comment by 4383** [comment]
Hello @rk-yen 

For your information, I completely rewrote the way we package niet (https://github.com/openuado/niet/commit/9435064ceba621da10cc68c55be3bc1d96199e10).

I published a new tag (https://github.com/openuado/niet/releases/tag/3.0.0) and a new version to pypi (https://pypi.org/project/niet/3.0.0/).

The previous issue related to version number seems solved:

```bash
$ python3.11 -m pip install niet
Collecting niet
  Downloading niet-3.0.0-py3-none-any.whl (13 kB)
Collecting pyyaml>=5.1
  Using cached PyYAML-6.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (757 kB)
Collecting jmespath>=0.9.4
  Using cached jmespath-1.0.1-py3-none-any.whl (20 kB)
Installing collected packages: pyyaml, jmespath, niet
Successfully installed jmespath-1.0.1 niet-3.0.0 pyyaml-6.0
$ niet --version
niet version 3.0.0
```

Please confirm that your are also able to install this new version without error.

Thanks

**Reply by rk-yen** [reply]
@4383 ... thank you!!

Question:
Is there a reason Python 3.8 is not supported?

**Comment by 4383** [comment]
Well, to be honest I think niet 3.0.0 work well with py3.8, I've not introduced changes that can broke the support of python 3.8, however, I've to reduce my maintenance scope to continue to maintain niet without pain and efforts, so I preferred to stick to the more recent versions of python.

Python 3.8 is still actively maintained
https://endoflife.date/python

**Reply by 4383** [reply]
@rk-yen FYI ^

**Reply by rk-yen** [reply]
Fair enough. Thank you.

**Comment by 4383** [comment]
Ok, thanks.
Then lets close this discussion.
